.github/workflows/build.yml

```diff
@@ -55,35 +60,40 @@ jobs:
           go-version-file: "go.mod"
           cache: false
 
+      - name: Install system dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y libsecret-1-dev
+
       - name: Build Linux AMD64
         run: make
         env:
           GOOS: linux
           GOARCH: amd64
-          CGO_ENABLED: 0
+          CGO_ENABLED: 1
```

Comment:
why does it need CGO now?

Reply:
this is a dependency coming from MSAL extension library we're now relying on for caching:
https://github.com/AzureAD/microsoft-authentication-extensions-for-go/tree/main/cache

This is the summary specific to each suppported OS:
```
Linux:
Requires CGO: Uses libsecret for secure storage
Dependencies: libsecret-1-dev package
CGO_ENABLED: 1


Windows:
Does NOT require CGO: Uses Windows Credential Manager via Windows APIs
Dependencies: None (uses built-in Windows APIs)
CGO_ENABLED: 0 (static binaries work fine)

macOS:
Requires CGO: Uses Keychain via Security framework
Dependencies: Xcode/macOS SDK (already available on macOS runners)
CGO_ENABLED: 1
```